pandas/tests/test_tseries.py

```diff
@@ -316,6 +316,13 @@ def test_maybe_booleans_to_slice():
     assert(result == slice(0, 0))
 
 
+def test_datetimelike_tostring_astype():
+    import pandas as pd
+    pd.date_range('2012-01-01', periods=4).astype(str)
+    result = ['2012-01-01', '2012-01-02', '2012-01-03', '2012-01-04']
+    assert(type(result[0]) == np.str)
```

Comment:
The test should compare the result and expected as below. I understand the result will be ``np.object`` dtype, not ``numpy.str``.

```
result = pd.date_range('2012-01-01', periods=4).astype(str)
expected = pd.Index(['2012-01-01', '2012-01-02', '2012-01-03', '2012-01-04'], dtype=object)
tm.assert_index_equal(result, expected)
```

Also, can you add other test cases, such as different freq like ``H``, and with timezone.